shark_turbine/kernel/wave/expansion.py

```diff
@@ -169,10 +169,10 @@ def set_node_index(
                     (
                         workgroup_constraints[dim].workgroup_dim,
                         (
-                            0
+                            1
                             if not is_contiguous_dim(
                                 dim,
-                                custom.type.symbolic_shape,
+                                custom.indexing_dims,
```

Comment:
Seems correct, but I would like to know how was it breaking before?

Reply:
Indexing dims for the read currently implemented like this:
```
    @property
    def indexing_dims(self) -> list[IndexSymbol]:
        if self.mapping is not None:
            return list(self.mapping.output_shape)
        # TODO: This could contain ints.
        return list(self.type.symbolic_shape)
```
If mapping is present we are using shape from mapping for indexing (output shape for read, input shape for write).
Without this fix `im2col` read will use `N, C, H, W` for indexing instead of expected `M, K`

Reply:
Ahhh, makes sense, thanks! :)